seleniumJSEnv/src/main/scala/org/scalajs/jsenv/selenium/SeleniumJSEnv.scala

```diff
@@ -2,35 +2,108 @@ package org.scalajs.jsenv.selenium
 
 import org.scalajs.core.tools.io.VirtualJSFile
 import org.scalajs.core.tools.jsdep.ResolvedJSDependency
-import org.scalajs.jsenv.{AsyncJSEnv, ComJSEnv}
-import org.scalajs.jsenv.{JSRunner, AsyncJSRunner, ComJSRunner}
 
-class SeleniumJSEnv private (browser: SeleniumBrowser, keepAlive: Boolean, materializer: FileMaterializer)
-    extends AsyncJSEnv with ComJSEnv {
+import org.openqa.selenium._
+import org.openqa.selenium.remote.DesiredCapabilities
+import org.openqa.selenium.remote.server._
+
+import org.scalajs.jsenv._
 
-  def this(browser: SeleniumBrowser) =
-    this(browser, keepAlive = false, materializer = DefaultFileMaterializer)
+import scala.reflect.{ClassTag, classTag}
 
-  def withMaterializer(materializer: FileMaterializer): SeleniumJSEnv =
-    new SeleniumJSEnv(browser, keepAlive, materializer)
+class SeleniumJSEnv(capabilities: Capabilities, config: SeleniumJSEnv.Config)
+    extends AsyncJSEnv with ComJSEnv {
 
-  def withKeepAlive(): SeleniumJSEnv =
-    new SeleniumJSEnv(browser, keepAlive = true, materializer)
+  def this(capabilities: Capabilities) =
+    this(capabilities, SeleniumJSEnv.Config())
 
-  def browserName: String = browser.name
+  private val augmentedCapabilities = {
+    val x = new DesiredCapabilities(capabilities)
+    x.setJavascriptEnabled(true)
+    x
+  }
 
-  def name: String = "SeleniumJSEnv for " + browserName
+  def name: String = s"SeleniumJSEnv ($capabilities)"
 
   def jsRunner(libs: Seq[ResolvedJSDependency], code: VirtualJSFile): JSRunner =
-    new SeleniumRunner(browser, libs, code, keepAlive, materializer)
+    new SeleniumRunner(newDriver _, libs, code, config)
 
   def asyncRunner(libs: Seq[ResolvedJSDependency],
       code: VirtualJSFile): AsyncJSRunner = {
-    new SeleniumAsyncJSRunner(browser, libs, code, keepAlive, materializer)
+    new SeleniumAsyncJSRunner(newDriver _, libs, code, config)
   }
 
   def comRunner(libs: Seq[ResolvedJSDependency],
       code: VirtualJSFile): ComJSRunner = {
-    new SeleniumComJSRunner(browser, libs, code, keepAlive, materializer)
+    new SeleniumComJSRunner(newDriver _, libs, code, config)
+  }
+
+  private def newDriver() = {
+    val driver: WebDriver =
+      config.driverFactory.newInstance(augmentedCapabilities)
+
+    /* The first `asInstanceOf`s are a fail-fast for the second one, which
+     * scalac partially erases, so that we're sure right now that the last
+     * cast is correct, as opposed to crashing when we call a method of
+     * JavascriptExecutor on the driver.
+     *
+     * We are "allowed" to cast since we explicitly request JavascriptEnabled in
+     * the capabilities.
+     */
+    driver.asInstanceOf[JavascriptExecutor]
+
+    driver.asInstanceOf[WebDriver with JavascriptExecutor]
+  }
+}
+
+object SeleniumJSEnv {
+  final class Config private (
+      val driverFactory: DriverFactory,
```

Comment:
Hum but now, what is this `DriverFactory`? Is the one that's in `private[selenium] object AbstractSeleniumJSRunner`? Because if yes, it leaks in a public API.